Sources/SwiftProtobuf/BinaryDecoder.swift

```diff
@@ -917,6 +917,12 @@ internal struct BinaryDecoder: Decoder {
                 try KeyType.decodeSingular(value: &k, from: &subdecoder)
             case 2: // Value is a message type
                 try subdecoder.decodeSingularEnumField(value: &v)
+                if v == nil {
```

Comment:
Not yet, but when dealing with #342 there could be; but that's a future PR.  I'm almost down doing tests to go with #334, so that will be a PR (likely tomorrow given the time here).